Requested Fulfillment union

ashkan18 opened the pull request:
# Problem
Follow up with Exchange changes in https://github.com/artsy/exchange/pull/129 where we introduced `requestedFulfillmentUnionType` for handling different types of requested fulfillments for an order.
We can now use fragment on this type to figure out if its shipping or pickup.
In case of `Ship` it includes all of shipping related fields without `shipping` prefix, so something like 

```graphql
requestedFulfillment {
    ... on Ship {
      name
      addressLine1
      addressLine2
      city
      region
      country
      postalCode
    }
  }
```

# Other Changes
Added missing `pageInfo` stuff to orders and line items fetching.

alloy commented on src/schema/ecommerce/types/requested_fulfillment_union_type.ts:
```diff
@@ -0,0 +1,56 @@
+export const RequestedFulfillmentUnionType = new GraphQLUnionType({
+  name: "RequestedFulfillment",
+  types: [Ship, Pickup],
+  resolveType: obj => (obj.country ? Ship : Pickup),
```
What about using `obj.__typename` for this? We moved away from type detection through fields, as it was a bit too brittle at times.

orta replied:
As long as we remember to include `__typename` in the query, that should be fine

orta commented on src/schema/ecommerce/query_helpers.ts:
```diff
@@ -0,0 +1,25 @@
+import gql from "lib/gql"
+
+export const RequestedFulfillmentFragment = gql`
```
and include it at the top, that's a good idea 👍

alloy replied:
In [another PR](
https://github.com/artsy/metaphysics/pull/1239#discussion_r212623281) we went for an anonymous inline fragment, because then you can just interpolate it inline.

ashkan18 commented:
addressed comments in ☝️ PR, still had issues with inline fragment but made small change there. @mzikherman maybe we can pair later on getting inline (or real) fragment working.
